rust-runtime/aws-smithy-async/src/rt/sleep.rs

```diff
@@ -83,7 +83,7 @@ pub fn default_async_sleep() -> Option<SharedAsyncSleep> {
 /// Future returned by [`AsyncSleep`].
 #[non_exhaustive]
 #[must_use]
-pub struct Sleep(Pin<Box<dyn Future<Output = ()> + Send + 'static>>);
+pub struct Sleep(Pin<Box<dyn Future<Output = ()> + Send + Sync + 'static>>);
```

Comment:
Futures shouldn't be `Sync`.

Reply:
After talking with Carl, I decided that this future should in fact be sync. While we could wrap this future in a mutex or similar to make it `Sync`, I'm opting to just add the bound. The reasoning is that omitting the `Sync` bound will cause more user pain than including it.

Reply:
Requiring a future to be `Sync` can cause some issue when used with an async block ([lang forum](https://users.rust-lang.org/t/future-is-not-sync-when-calling-async-fn-of-dyn-trait-obj-in-another-async-fn/83285)), but I don't necessarily see how that bites us here 🤔

At least, I think this PR needs a label `breaking-change` because `Sleep::new` now fails to take an async block for certain cases ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=59ab7e7a4f8aead7788484a1b1fb339c) - derived from the above lang forum).